sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -35,6 +35,20 @@
 from sklearn.utils.testing import assert_array_equal
 from sklearn.utils.testing import assert_warns
 
+def test_deprecation_of_n_components_in_linkage_tree():
+
+    for tree_builder in _TREE_BUILDERS.values():
+        for this_X in (X, X[0]):
+            with ignore_warnings():
+                    # Test for warning of deprecation of n_components in linkage_tree
+                    children, n_nodes, n_leaves, parent = assert_warns(DeprecationWarning, linkage_tree,this_X.T, n_components=10)
+                    children1, n_nodes1, n_leaves1, parent1 = linkage_tree(this_X.T, n_components=10)
```

Comment:
The previous line will calculate the same. As long as we believe assert_warns works, and we do